pyof/foundation/basic_types.py

```diff
@@ -421,10 +421,23 @@ def __init__(self, value=b''):  # noqa
         Raises:
             ValueError: If given value is not bytes.
         """
-        if not isinstance(value, bytes):
-            raise ValueError('BinaryData must contain bytes.')
+        value = self._pack_if_necessary(value)
         super().__init__(value)
 
+    @staticmethod
+    def _pack_if_necessary(value):
```

Comment:
I think leaving only _pack makes one think that it will pack anyway; furthermore, the ``_pack`` method was used for another purpose in the original PR from which this commit came.